codegen/aws/core/src/main/java/software/amazon/smithy/python/aws/codegen/AwsUserAgentIntegration.java

```diff
@@ -0,0 +1,104 @@
+                            .definitionFile(String
+                                    .format("./%s/%s.py", moduleName, user_agent_plugin_file))
+                            .name("aws_user_agent_plugin")
+                            .build())
+                    .build();
+            final SymbolReference userAgentInterceptor = SymbolReference.builder()
+                    .symbol(Symbol.builder()
+                            .namespace("smithy_aws_core.interceptors.user_agent", ".")
+                            .name("UserAgentInterceptor")
+                            .build())
+                    .build();
+            final SymbolReference versionSymbol = SymbolReference.builder()
+                    .symbol(Symbol.builder()
+                        .namespace(moduleName, ".")
+                        .name("__version__")
+                        .build()
+                    ).build();
+
+            return List.of(
+                    RuntimeClientPlugin.builder()
+                            .addConfigProperty(userAgentExtra)
+                            .addConfigProperty(uaAppId)
+                            .pythonPlugin(userAgentPlugin)
+                            .writeAdditionalFiles((c) -> {
+                                String filename = "%s/%s.py".formatted(moduleName, user_agent_plugin_file);
+                                c.writerDelegator()
+                                        .useFileWriter(
+                                                filename,
+                                                moduleName + ".",
+                                                writer -> {
+                                                    writer.write("""
+                                                            def aws_user_agent_plugin(config: $1T):
+                                                                config.interceptors.append(
+                                                                    $2T(
+                                                                        ua_suffix=config.user_agent_extra,
+                                                                        ua_app_id=config.sdk_ua_app_id,
+                                                                        sdk_version=$3T,
+                                                                        service_id='$4L'
+                                                                    )
+                                                                )
+                                                            """,
+                                                            CodegenUtils.getConfigSymbol(c.settings()),
+                                                            userAgentInterceptor,
+                                                            versionSymbol,
+                                                            c.settings().service().getName()
+                                                    );
```

Comment:
Why did you move this into Java?

Reply:
To get the client library version and service id - since both of those are specific to the generated client.

The alternative would be to add that information to the interceptor context (or potentially to use some python metaprogramming hacks?). 

Do you have thoughts or preferences between those? (or maybe another obvious way I'm missing to get client library version and service id)

Reply:
Ah yeah the service id isn't available right now because we're not writing out service schemas, and thus those traits aren't present.